cli/src/commands/explore/runtime_apis/mod.rs

```diff
@@ -0,0 +1,204 @@
+///                               response is:
+///                                 Err => Show Error
+///                                 Ok  => Show the result
+/// ```
+pub async fn run<'a>(
+    method: Option<String>,
+    execute: bool,
+    trailing_args: Vec<String>,
+    runtime_api_metadata: RuntimeApiMetadata<'a>,
+    metadata: &'a Metadata,
+    file_or_url: FileOrUrl,
+    output: &mut impl std::io::Write,
+) -> color_eyre::Result<()> {
+    let api_name = runtime_api_metadata.name();
+
+    let usage = || {
+        let methods = methods_to_string(&runtime_api_metadata);
+        formatdoc! {"
+        Usage:
+            subxt explore api {api_name} <METHOD>
+                explore a specific runtime api method
+        
+        {methods}
+        "}
+    };
+
+    // If method is None: Show pallet docs + available methods
+    let Some(method_name) = method else {
+        let doc_string = first_paragraph_of_docs(runtime_api_metadata.docs()).indent(4);
+        if !doc_string.is_empty() {
+            writedoc! {output, "
+            Description:
+            {doc_string}
+    
+            "}?;
+        }
+        writeln!(output, "{}", usage())?;
+        return Ok(());
+    };
+
+    // If method is invalid: Show Error + available methods
+    let Some(method) = runtime_api_metadata
+        .methods()
+        .find(|e| e.name().eq_ignore_ascii_case(&method_name))
+    else {
+        return Err(eyre!(
+            "\"{method_name}\" method not found for \"{method_name}\" runtime api!\n\n{}",
+            usage()
+        ));
+    };
+    // redeclare to not use the wrong capitalization of the input from here on:
+    let method_name = method.name();
+
+    // Method is valid. Show method docs + output type description
+    let doc_string = first_paragraph_of_docs(method.docs()).indent(4);
+    if !doc_string.is_empty() {
+        writedoc! {output, "
+        Description:
+        {doc_string}
+        
+        "}?;
+    }
+
+    #[allow(non_snake_case)]
+    let INPUT_VALUE: String = "<INPUT_VALUE>".blue().to_string();
+
+    // Output type description
+    let input_values = || {
+        if method.inputs().len() == 0 {
+            return format!("The method does not require an {INPUT_VALUE}");
+        }
+
+        let fields: Vec<(Option<&str>, u32)> = method
+            .inputs()
+            .map(|f| (Some(f.name.as_str()), f.ty))
+            .collect();
+        let fields_description =
+            fields_description(&fields, method.name(), metadata.types()).indent(4);
+
+        let fields_example =
+            fields_composite_example(method.inputs().map(|e| e.ty), metadata.types())
+                .indent(4)
+                .highlight();
+
+        formatdoc! {"
+        The method expects an {INPUT_VALUE} with this shape:
+        {fields_description}
+    
+        For example you could provide this {INPUT_VALUE}:
+        {fields_example}"}
+    };
+
+    let execute_usage = || {
```

Comment:
This could just be a block?

Reply:
Oh, good catch, I made it a closure to use it twice below, but used it only once. I am now replacing the `usage()` below by the second `execute_usage()` as intended.